sdk/python/kfp/components/_components.py

```diff
@@ -196,18 +196,27 @@ def _react_to_incompatible_reference_type(
     raise TypeError(message)
 
 
-def _create_task_spec_from_component_and_arguments(
-    component_spec: ComponentSpec,
+def _create_task_spec_from_task_spec_and_arguments(
+    task_spec: TaskSpec,
     arguments: Mapping[str, Any],
-    component_ref: ComponentReference = None,
 ) -> TaskSpec:
-    """Constructs a TaskSpec object from component reference and arguments.
-    The function also checks the arguments types and serializes them."""
-    if component_ref is None:
-        component_ref = ComponentReference(spec=component_spec)
-    else:
-        component_ref = copy.copy(component_ref)
-        component_ref.spec = component_spec
+    """Constructs a task object from an argument-less TaskSpec and arguments.
+    
+    This function provides a reference implementation for the _default_container_task_constructor that returns TaskSpec objects.
+    The only reference-type arguments that TaskSpec can hold are TaskOutputArgument and GraphInputArgument.
+    When bridging with other frameworks, an alternative implementation should be provided that can process reference-type arguments that are native to that framework.
+    This function also checks the arguments types and serializes constant arguments.
+
+    Args:
+        task_spec: Argument-less TaskSpec object that holds component_spec and other options, but not arguments which are passed separately. Should have task_spec.component_ref.spec != None
+        arguments: Map with arguments passed to the component. Keys should correspond to the input names. For this implementation, the values can be constant values, TaskOutputArgument or GraphInputArgument
+
+    Returns:
+        TaskSpec object with populated arguments.
+    """
+    component_spec = task_spec.component_ref.spec
```

Comment:
qq: does `task_spec` always have component_ref?

Reply:
Yes. `component_ref` is the only required field of `TaskSpec`.
https://github.com/kubeflow/pipelines/blob/39975e3cde7ba4dcea2bca835b92d0fe40b1ae3c/sdk/python/kfp/components/_structures.py#L571